### PR TITLE
fix server deck storage; fix #744

### LINKED ISSUE #744: Server Deck Storage severely bugged
Yesterday, I created 3 folders for decks. Modern, EDH, and Other. I log on today, only Modern is there. I delete this folder. Then, after the server went down for maintenance, the EDH folder was there (still no Other, though).

I deleted the folders because I don't really use server storage anyway, but that's really bugged and doesn't seem good at all.

### PR DESCRIPTION
**Pull request by ctrlaltca**

### CODE REVIEW

File: servatrice/src/serversocketinterface.cpp [hunk 1]
```diff
@@ -375,10 +375,15 @@ bool ServerSocketInterface::deckListHelper(int folderId, ServerInfo_DeckStorage_
     if (!sqlInterface->execSqlQuery(query))
         return false;
 
-    while (query->next()) {
+    QMap<int, QString> results;
+    while(query->next())
+        results[query->value(0).toInt()] = query->value(1).toString();
+
+    foreach(int key, results.keys())
```
Comment (Daenyth) [inline servatrice/src/serversocketinterface.cpp]: I thought we couldn't use this yet?

Comment (ctrlaltca) [inline servatrice/src/serversocketinterface.cpp]: We already use it in a lot of places in the code. It's a macro defined by Qt.

**Comment by woogerboy21** [comment]
IS there a way to kick travis off again?  
```
No output has been received in the last ... minutes, this potentially indicates a stalled build or something wrong with the build itself.

The build has been terminated
```

**Comment by ctrlaltca** [comment]
relaunched, now working.

**Pull request merged**